### DISCUSSION TITLE
Looking good! Got some feedback

### DISCUSSION THREAD

**Discussion by ivancuric** [daniel-nagy/transporter#7] [General]
Hey Daniel, wanted to say that this looks pretty good.

Looking at the API I have two concerns:

1.  A lot of `enum` usage — not really a fan of `enum`s, especially in the case of libraries. Most of these values can be simple string unions, without forcing the user to essentially import multiple strings from 3rd party dependency. `enum`s also needlessly bloat the output bundle as you are using regular, not `const enum`s.
2. Requiring the user to write functions in a way that they must return a promise. This can mess up typechecking and eslint-typescript. Why not promisify the functions when passed to transporter?

**Comment by daniel-nagy** [comment]
@ivancuric Sorry for the delayed reply. I haven't been getting notifications from GitHub for activity on this repo for some reason.

> A lot of enum usage — not really a fan of enums

This seems pretty minor; I'd like to get more feedback from folks about how they feel about enums. Personally, I like enums. The core package only has 8 enums, and, most of the time, you will not need to import them.

> Requiring the user to write functions in a way that they must return a promise. This can mess up typechecking and eslint-typescript. Why not promisify the functions when passed to transporter?

I actually didn't require functions to be async initially, and I did promisify the functions. The reason I backed out of this was because it significantly increased the complexity of the TypeScript types. In addition, there were issues modifying the call signature of generic functions at the type level. If you modify the call signature of a generic function at the type level, you lose generics.
